src/helpers/TypeHelpers.ts

```diff
@@ -26,30 +24,30 @@ export class TypeHelpers {
   }
 }
 
-export type TypeContext<T extends ConstrainedMetaModel, R extends AbstractRenderer> = {
+export type TypeContext<T extends ConstrainedMetaModel, Options> = {
   propertyKey?: string,
-  renderer: R,
+  options: Options,
   constrainedModel: T,
 }
 
-export type TypeMappingFunction<T extends ConstrainedMetaModel, R extends AbstractRenderer> = (context: TypeContext<T, R>) => string;
+export type TypeMappingFunction<T extends ConstrainedMetaModel, Options> = (context: TypeContext<T, Options>) => string;
 
-export type TypeMapping<R extends AbstractRenderer> = {
-  Object: TypeMappingFunction<ConstrainedObjectModel, R>,
-  Reference: TypeMappingFunction<ConstrainedReferenceModel, R>,
-  Any: TypeMappingFunction<ConstrainedAnyModel, R>,
-  Float: TypeMappingFunction<ConstrainedFloatModel, R>,
-  Integer: TypeMappingFunction<ConstrainedIntegerModel, R>,
-  String: TypeMappingFunction<ConstrainedStringModel, R>,
-  Boolean: TypeMappingFunction<ConstrainedBooleanModel, R>,
-  Tuple: TypeMappingFunction<ConstrainedTupleModel, R>,
-  Array: TypeMappingFunction<ConstrainedArrayModel, R>,
-  Enum: TypeMappingFunction<ConstrainedEnumModel, R>,
-  Union: TypeMappingFunction<ConstrainedUnionModel, R>,
-  Dictionary: TypeMappingFunction<ConstrainedDictionaryModel, R>
+export type TypeMapping<Options> = {
```

Comment:
In `ConstrainContext` we have `Options` as first generic type, maybe we should also standarize it, and change here order?

```ts
export type TypeContext<Options, T extends ConstrainedMetaModel>
export type TypeMappingFunction<Options, T extends ConstrainedMetaModel>
```

? I'm just thinking out loud 😄

Reply:
I mean, aren't they in the same order currently? Just switched from what you suggested, of course `TypeMapping<Options>` is not, but I think that's okay.

Reply:
Yeah, but I refer to the `ConstrainContext` where `Options` is first argument, rest types have `Options` as last argument :P

Reply:
I see what you mean. Hmm, but would that not also be the case the other way around if you had a function that took a generic model type but a specific option? I.e. flipped case then here.

So does not really matter about changing it as it would be like potato potato 😆

Reply:
Yeah, right. Model has much "power" in this case, so it can stay as it is.